src/interactivity/Interactivity.js

```diff
@@ -147,22 +148,34 @@ export default class Interactivity {
         if (!map.__carto_interactivities) {
             map.__carto_interactivities = new Set();
         }
-        this.on('featureHover', event => {
-            if (event.features.length) {
-                map.__carto_interactivities.add(this);
-            } else {
-                map.__carto_interactivities.delete(this);
-            }
-            map.getCanvas().style.cursor = (map.__carto_interactivities.size > 0) ? 'pointer' : '';
-        });
+        this._onFeatureHoverBound = this._onFeatureHover.bind(this, map);
```

Comment:
I think we should avoid declaring the function we're going to use to listen the event, which is used in `_onLayerRemoved` where we're not checking if it's defined.

Instead, I would save the `map` in a "private" value for the Interactivity (`this._map = layerList[0].map`) and use it on `_onFeatureHover` method instead of passing it as a parameter, so we could use this function in `_onLayerRemoved` method. We wouldn't need then `_onFeatureHoverBound`.

Reply:
Yes, I thought it too when doing my own code review and it's better to change it like you said. We're checking if the auto cursor option is enabled, but it's better to hold the map in a private variable :)